### PR TITLE
Feature: Telemetry

### PR DESCRIPTION
**Pull request by onlyjackfrost**
### Description
This PR handle the launcher to pass unique user uuid and Integrated with Posthog as our telemetry tool

user Id and api keys will be passed by env.

### Test on local

Ask User can collect or not

Yes 🆗 
image

No 🚫 
image

Telemetry can receive events with corresponding system info
image

### CODE REVIEW

**Comment by wwwy3y3** [comment]
Adding a rcfile package that I can directly use to append configuration and also read current configs without knowing the path of the rcfile, format and worrying if I override other values.

@onlyjackfrost

File: Telemetry.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+
+### Telemetry
+
+WrenAI relies on anonymous usage statistics to continuously improve. That's why we ask you to share your usage data with us.
+WrenAI use [Posthog](https://posthog.com/) as out telemetry tool
```
Comment (cyyeh) [inline Telemetry.md:5]:
```markdown
WrenAI uses [Posthog](https://posthog.com/) as out telemetry tool.
```

File: Telemetry.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+
+### How Does Telemetry help WrenAI
+
+Based on the telemetry, we can know which features are frequently used and related to our community or in which step an error occured and block our user.
```
Comment (cyyeh) [inline Telemetry.md:10]: Based on the telemetry, we can know what features are frequently used and related to our community or in what step an error occurs and blocks our user.

File: Telemetry.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+...
+
+Here is an example event that we will collect:
+```
+# example to an "server start" event
+{
+  "uuid": "018eeb03-85db-73be-b7b6-84b2d735f19c", # the event id
+  "event": "server_start", 
+  "properties": {
+    "wren-ui-version": "telemetry",
```
Comment (cyyeh) [inline Telemetry.md:28]: values of wren-ui-version and wren-engine_version are strange. also there is underscore in `wren-engine_version`; while there is no underscore in `wren-ui-version`
 
  "wren-ui-version": "telemetry",
  "wren-engine_version": null,

File: Telemetry.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    "time": "2024-04-17T07:44:20.432Z",
+    "$lib": "posthog-node",
+    "$lib_version": "4.0.0",
+    "$geoip_disable": true,
+    "$sent_at": "2024-04-17T07:44:30.445000+00:00"
+  },
+  "timestamp": "2024-04-17T07:44:21.060000Z",
+  "team_id": 63160,
+  "distinct_id": "4007a326-d606-460e-9acc-22f40cf851fb", # randomly generated userId
+  "elements_chain": "",
```
Comment (cyyeh) [inline Telemetry.md:54]: what is elements_chain? maybe we need to explain to users what does this mean or give them a document link from PostHog?

**Review by cyyeh — COMMENTED** [review]
some comments on Telemetry.md

File: Telemetry.md (hunk 4 above)
Comment (onlyjackfrost) [reply Telemetry.md:54]: Do you find any doc that describe this in posthog?

Comment (cyyeh) [reply Telemetry.md]: https://www.perplexity.ai/search/What-is-element-TE5KCPYITzCfdxy94Qw88A#0

**Review by wwwy3y3 — APPROVED** [review]
LGTM

File: wren-ui/src/apollo/server/services/askingService.ts [hunk 5]
```diff
@@ -187,6 +192,10 @@ class BackgroundTracker {
 
           // remove the task from tracker if it is finalized
           if (isFinalized(result.status)) {
+            this.telemetry.send_event('question_answered', {
+              question: threadResponse.question,
+              result,
```
Comment (wwwy3y3) [inline wren-ui/src/apollo/server/services/askingService.ts:197]: image

`Results` is like this. No personal information included checked.

**Review by wwwy3y3 — APPROVED** [review]
LGTM

**Pull request merged**